pydantic_ai_slim/pydantic_ai/models/cohere.py

```diff
@@ -234,6 +234,9 @@ def _map_messages(
     ) -> list[ChatMessageV2]:
         """Just maps a `pydantic_ai.Message` to a `cohere.ChatMessageV2`."""
         cohere_messages: list[ChatMessageV2] = []
+        system_prompt_count = sum(
+            1 for m in messages if isinstance(m, ModelRequest) for p in m.parts if isinstance(p, SystemPromptPart)
```

Comment:
This will result in a wrong index when the system prompt parts are not all of the beginning, and it seems brittle to use an index derived from Pydantic AI `messages` in a list named `cohere_messages`, which may or may not map 1:1. So I think we should count system parts at the start of the actual `cohere_messages` list instead